tests/rust/aligned_struct.rs

```diff
@@ -0,0 +1,84 @@
+#[repr(packed, C)]
+pub struct PackedStruct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+#[repr(align(1), C)]
+pub struct Align1Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+#[repr(align(2), C)]
+pub struct Align2Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+#[repr(align(4), C)]
+pub struct Align4Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+#[repr(align(8), C)]
+pub struct Align8Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+#[repr(align(32), C)]
+pub struct Align32Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+// Packed doesn't take an argument.
+#[repr(packed(4), C)]
+pub struct BadPacked4Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+// Alignment must be an integer.
+#[repr(align("1"), C)]
+pub struct BadAlign1Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+// Alignment must be an integer.
+#[repr(align(foobar), C)]
+pub struct BadAlignFoobarStruct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+// Alignment must be a power of two.
+#[repr(align(3), C)]
+pub struct BadAlign3Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+// Alignment must be a power of two.
+#[repr(align(6), C)]
+pub struct BadAlign6Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+// Alignment takes exactly one argument.
+#[repr(align, C)]
+pub struct BadAlignStruct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+// Alignment takes exactly one argument.
+#[repr(align(2, 4), C)]
+pub struct BadAlign2_4Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
```

Comment:
Let's add also tests for only having repr(align) and such, and not having `repr(C)`. Those shouldn't be generated.